lib/helpers/preserve_data.rb

```diff
@@ -0,0 +1,58 @@
+require 'fileutils'
+require 'json'
+
+module PreserveData
+  def save_collections
+    save_albums
+    save_genres
+  end
+
+  def load_collections
+    load_albums
+    load_genres
+  end
+
+  private
+
+  def save_albums
+    base_folder = './lib/helpers/json'
+    albums_path = './lib/helpers/json/albums.json'
+
+    FileUtils.mkdir_p(base_folder) unless Dir.exist?(base_folder)
+    json_albums = @albums.map { |album| album.to_json }
+    File.write(albums_path, JSON.generate(json_albums))
+  end
+
+  def save_genres
+    base_folder = './lib/helpers/json'
+    genres_path = './lib/helpers/json/genres.json'
+
+    FileUtils.mkdir_p(base_folder) unless Dir.exist?(base_folder)
+    json_genres = @genres.map { |genre| genre.to_json }
+    File.write(genres_path, JSON.generate(json_genres))
+  end
```

Comment:
This code (line21 and 30) it's checking for json folder in all 'save' methods, maybe we can use it just once?

Reply:
done! thanks